### ISSUE TITLE
[BUG] JSQLParser 4.6: NPE on Postgres SELECT with FILTER

### ISSUE THREAD

**Issue by sambsnyd** [JSQLParser/JSqlParser#1850]
### Failing SQL Feature:
Given a query where a SELECT statement includes a FILTER I get an NPE:
```
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because the return value of "net.sf.jsqlparser.expression.AnalyticExpression.getOrderByElements()" is null
	at net.sf.jsqlparser.expression.ExpressionVisitorAdapter.visit(ExpressionVisitorAdapter.java:342)
	at net.sf.jsqlparser.expression.AnalyticExpression.accept(AnalyticExpression.java:73)
	at net.sf.jsqlparser.expression.ExpressionVisitorAdapter.visit(ExpressionVisitorAdapter.java:523)
	at net.sf.jsqlparser.statement.select.SelectExpressionItem.accept(SelectExpressionItem.java:46)
```

It looks like there is a missing null guard in `ExpressionVisitorAdapter.visit(AnalyticExpression)`:
```java
    @Override
    public void visit(AnalyticExpression expr) {
        ...
        // NPE here because expr.getOrderByElements() returns null
        for (OrderByElement element : expr.getOrderByElements()) {
            element.getExpression().accept(this);
        }

        ...
    }
```
I don't know if fixing this NPE is the only thing preventing this from parsing successfully or not
### SQL Example:
```
SELECT COUNT(*) FILTER (WHERE state  = 'CANCELED')
                        FROM commit
```

### Software Information:
- JSqlParser version 4.6
- Database Postgres

**Comment by manticore-projects** [comment]
Greetings.

Thank you for reporting. You are right:
```java
private List<OrderByElement> funcOrderBy = null;
```

warrants a Null check.

**Comment by manticore-projects** [comment]
Fixed via beb68d55239da97ca2ce8f513212740ebe94ce81
